src/nti/fakestatsd/__init__.py

```diff
@@ -7,136 +7,5 @@
-
-    See also: https://github.com/etsy/statsd/blob/master/docs/metric_types.md
-    """
-    metrics = []
-
-    # Start by splitting newlines. this may be a multi-metric packet
-    for metric_data in data.split('\n'):
-        sampling = None
-        name = None
-        value = None
-        kind = None
-
-        # Metrics take the form of <name>:<value>|<type>(|@<sampling_rate>)
-        parts = metric_data.split('|')
-        if len(parts) < 2 or len(parts) > 3:
-            raise ValueError('Unexpected metric data %s. Wrong number of parts' % metric_data)
-
-        if len(parts) == 3:
-            sampling_data = parts.pop(-1)
-            sampling = _parse_sampling_data(sampling_data)
-
-        kind = parts[1]
-        name, value = parts[0].split(':')
-
-        metrics.append(METRIC_TYPES[kind](name, value, sampling, kind))
-    return metrics
-
-
-class FakeStatsDClient(StatsdClient):
-    """
-    A mock statsd client that tracks sent statsd metrics in memory
-    rather than pushing them over a socket.
-    """
-
-    def __init__(self, prefix=''):
-        """
-        Create a mock statsd client with the given prefix.
-        """
-        super(FakeStatsDClient, self).__init__(prefix=prefix)
-
-        # Monkey patch the socket to track things in memory instead
-        self._raw = _raw = []
-        self._metrics = _metrics = []
-
-        class DummySocket(object):
-            def sendto(self, data, addr):
-                # The client encoded to bytes
-                assert isinstance(data, bytes)
-                # We always want native strings here, that's what the
-                # user specified when calling the StatsdClient methods.
-                data = data.decode('utf-8') if bytes is not str else data
-                _raw.append((data, addr,))
-                for m in _as_metrics(data):
-                    _metrics.append((m, addr,))
-
-        self.udp_sock.close()
-        self.udp_sock = DummySocket()
-
-    def clear(self):
-        """
-        Clears the statsd metrics that have been collected
-        """
-        del self._raw[:]
-        del self._metrics[:]
-
-    def __len__(self):
-        """
-        The number of metrics sent. This accounts for multi metric packets
-        that may be sent.
-        """
-        return len(self._metrics)
-
-    def __iter__(self):
-        """
-        Iterates the metrics provided to this statsd client
-        """
-        for metric, _ in self._metrics:
-            yield metric
-    iter_metrics = __iter__
-
-    def iter_raw(self):
-        """
-        Iterates the raw metrics provided to the stats d client.
-        """
-        for data, _ in self._raw:
-            yield data
-
-    @property
-    def metrics(self):
-        return [m for m in self]
-
-    @property
-    def packets(self):
-        return [p for p in self.iter_raw()]
+from .client import FakeStatsDClient
+from .metric import Metric
```

Comment:
All modules that we intend to be public should specify an `__all__` attribute detailing what's public. In this case:
```python
__all__ = [
    'FakeStatsDClient',
    'Metric',
]